src/dstore/pmix_esh.c

```diff
@@ -2203,24 +2203,21 @@ static int pmix_sm_store(ns_track_elem_t *ns_info, pmix_rank_t rank, pmix_kval_t
 
     datadesc = ns_info->data_seg;
     /* pack value to the buffer */
-    buffer = PMIX_NEW(pmix_buffer_t);
-    if (PMIX_SUCCESS != (rc = pmix_bfrop.pack(buffer, kval->value, 1, PMIX_VALUE))) {
-        PMIX_RELEASE(buffer);
-        PMIX_ERROR_LOG(rc);
-        return rc;
+    PMIX_CONSTRUCT(&buffer, pmix_buffer_t);
+    if (PMIX_SUCCESS != (rc = pmix_bfrop.pack(&buffer, kval->value, 1, PMIX_VALUE))) {
```

Comment:
Unfortunately we need `PMIX_ERROR_LOG(PMIX_ERROR);` here otherwise line number will be less representative.
As I've said it won't save space - only ensure proper cleanup.